lib/valid_email2/address.rb

```diff
@@ -130,10 +137,17 @@ def mx_server_is_in?(domain_list)
       }
     end
 
-    def address_contain_emoticons?
+    def address_contain_multibyte_characters?
       return false if @raw_address.nil?
 
-      @raw_address.scan(Unicode::Emoji::REGEX).length >= 1
+      multibyte_chars = @raw_address.scan(/./).select { |char| char.bytesize > 1 }
+      return false if multibyte_chars.empty?
+
+      return true if self.class.permitted_multibyte_characters_regex.nil?
+
+      multibyte_chars.any? do |char|
+        !self.class.permitted_multibyte_characters_regex.match?(char)
+      end
```

Comment:
```suggestion
      return false if @raw_address.ascii_only?

      @raw_address.each_char.any? { |char| char.bytesize > 1 && char !~ self.class.permitted_multibyte_characters_regex }
```

This reads cleaner to me, and we return early without having to resort to scanning the whole string and checking each characters bytesize.

Reply:
I didn't know about the `ascii_only?` method.
It was very clear! Thanks.